_docs/koppelaal-1.2/write-connector.md

```diff
@@ -0,0 +1,125 @@
+
+_Note_: The Koppeltaal server supports both JSON and XML.
+
+![figure 1]
+![placeholder]
+**Figure 1**. _Each application communicates with the Koppeltaal server through a connector to encapsulate most of the complexity._
+
+Obviously, the functionalities a connector should provide depends on the applications that use it. However, we think there is a set of basic functionalities that a connector in general should support.
+- Retrieve the `Conformance Statement`
+- Send/retrieve `Messages` to/from a domain
+-- Create a Message bundle with FHIR Resources (send)
+-- Extract FHIR Resources from the Message bundle (receive)
+-- Update the _ProcessingStatus_ of a Message
+- Retrieve and update `Activity Definitions`
+
+Additionally, a connector could also support the following functionality:
+- Create a RESTful endpoint and configure a webhook on KT server
+- Using the `Storage Service` as data persistance utitliy
+
+The connector will achieve most of this functionality by sending/retrieving Messages according to Koppeltaal specifications.
+view of the infrastructure entities that the connector will be dealing with.
+
+### Basic features
+
+#### The Conformance Statement
+A `Conformance Statement` is a set of capabilities of a FHIR Server. The conformance statement of the Koppeltaal Server contains information required for the OAuth2 implementation for the single sign on. Besides the capabilities, it contains a number of important endpoints that are needed for certain actions. It can be requested via _<Koppeltaal server url>/FHIR/Koppeltaal/metadata_ (e.g. http://edgekoppeltaal.vhscloud.nl/FHIR/Koppeltaal/metadata).
+The connector must extract these endpoints from the Conformance statement and use them for other actions. For instance, to request authorization for the launch of a game, the authorization endpoint must be read from the statement.
+
+#### Koppeltaal Messages
+Communication with the Koppeltaal server consists largely of exchanging `Messages`. The Koppeltaal message consists of a bundle of `FHIR Resources`. More information on the [FHIR resources] used in Koppeltaal.
+The first resource in the bundle is the `MessageHeader`, which should always be present. The header contains important information about the content of the message, like the [message event types]. This information can be used to convert the resources in the bundle to a format that can be further processed by an application - e.g. an object holding all the important values to an application.
+The relevant values of the Resource should be copied to simple objects that can be handled by the application. It is possible that HL7 provides a library that provides FHIR models and validation schema's: [DSTU1] / [current version].
```

Comment:
"e.g." means it's an example, it is indeed dependent on the programming language.

Also working out your last comment for the pull request next time.